<h1 align="center">0x05.PYTHON - EXCEPTIONS</h1>

Here is a summary of what I learnt in this project: 
<ul>
<li>How to use <code>try: / except:</code> to handle exception.</li>
<li>How to use <code>raise</code> to raise an exception.</li>
</ul>

|File|Description|
|--|--|
|[0-safe_print_list.py](https://github.com/GM-Samuelstein/alx-higher_level_programming/blob/master/0x05-python-exceptions/0-safe_print_list.py)|A function that prints x elements of a list.|
|[0-main.py](https://github.com/GM-Samuelstein/alx-higher_level_programming/blob/master/0x05-python-exceptions/0-main.py)|A python file that tests the function: <code>def safe_print_list(my_list=[], x=0):</code> defined in 0-safe_print_list.py.|
|[1-safe_print_integer.py](https://github.com/GM-Samuelstein/alx-higher_level_programming/blob/master/0x05-python-exceptions/1-safe_print_integer.py)|A function that prints an integer with <code>"{:d}".format()</code>.|
|[1-main.py](https://github.com/GM-Samuelstein/alx-higher_level_programming/blob/master/0x05-python-exceptions/1-main.py)|A python file that tests the function: <code>def safe_print_integer(value):</code> defined in 1-safe_print_integer.py.|
|[2-safe_print_list_integers.py](https://github.com/GM-Samuelstein/alx-higher_level_programming/blob/master/0x05-python-exceptions/2-safe_print_list_integers.py)|A function that prints the first x elements of a list and only integers.|
|[2-main.py](https://github.com/GM-Samuelstein/alx-higher_level_programming/blob/master/0x05-python-exceptions/2-main.py)|A python file that tests the function: <code>def safe_print_list_integers(my_list=[], x=0):</code> defined in 2-safe_print_list_integers.py.|
|[3-safe_print_division.py](https://github.com/GM-Samuelstein/alx-higher_level_programming/blob/master/0x05-python-exceptions/3-safe_print_division.py)|A function that divides 2 integers and prints the result.|
|[3-main.py](https://github.com/GM-Samuelstein/alx-higher_level_programming/blob/master/0x05-python-exceptions/3-main.py)|A python file that tests the function: <code>def safe_print_division(a, b):</code> defined in 3-safe_print_division.py.|
|[4-list_division.py](https://github.com/GM-Samuelstein/alx-higher_level_programming/blob/master/0x05-python-exceptions/4-list_division.py)|A function that divides element by element 2 lists.|
|[4-main.py](https://github.com/GM-Samuelstein/alx-higher_level_programming/blob/master/0x05-python-exceptions/4-main.py)|A python file that tests the function: <code>def list_division(my_list_1, my_list_2, list_length):</code> defined in 4-list_division.py.|
|[5-raise_exception.py](https://github.com/GM-Samuelstein/alx-higher_level_programming/blob/master/0x05-python-exceptions/5-raise_exception.py)|A function that raises a type exception.|
|[5-main.py](https://github.com/GM-Samuelstein/alx-higher_level_programming/blob/master/0x05-python-exceptions/5-main.py)|A python file that tests the function: <code>def raise_exception():</code> defined in 5-raise_exception.py.|
|[6-raise_exception_msg.py](https://github.com/GM-Samuelstein/alx-higher_level_programming/blob/master/0x05-python-exceptions/6-raise_exception_msg.py)|A function that raises a name exception with a message.|
|[6-main.py](https://github.com/GM-Samuelstein/alx-higher_level_programming/blob/master/0x05-python-exceptions/6-main.py)|A python file that tests the function: <code>def raise_exception_msg(message=""):</code> defined in 6-raise_exception_msg.py.|
|[100-safe_print_integer_err.py](https://github.com/GM-Samuelstein/alx-higher_level_programming/blob/master/0x05-python-exceptions/100-safe_print_integer_err.py)||
|[100-main.py](https://github.com/GM-Samuelstein/alx-higher_level_programming/blob/master/0x05-python-exceptions/100-main.py)||
|[101-safe_function.py](https://github.com/GM-Samuelstein/alx-higher_level_programming/blob/master/0x05-python-exceptions/101-safe_function.py)||
|[101-main.py](https://github.com/GM-Samuelstein/alx-higher_level_programming/blob/master/0x05-python-exceptions/101-main.py)||
|[102-magic_calculation.py](https://github.com/GM-Samuelstein/alx-higher_level_programming/blob/master/0x05-python-exceptions/102-magic_calculation.py)|A python function that does exactly the same as a given python bytecode.|
|[103-python.c](https://github.com/GM-Samuelstein/alx-higher_level_programming/blob/master/0x05-python-exceptions/103-python.c)|Three C functions that print some basic info about Python lists, Python bytes an Python float objects.|
